elide-datastore/elide-datastore-hibernate/src/main/java/com/yahoo/elide/core/filter/CaseAwareJPQLGenerator.java

```diff
@@ -71,25 +72,26 @@ public CaseAwareJPQLGenerator(String jpqlTemplate, Case upperOrLower, ArgumentCo
     }
 
     @Override
-    public String generate(String columnAlias, List<FilterPredicate.FilterParameter> params) {
+    public String generate(FilterPredicate predicate, Function<Path, String> aliasGenerator) {
+       String columnAlias = aliasGenerator.apply(predicate.getPath());
 
         if (Strings.isNullOrEmpty(columnAlias)) {
             log.error("columnAlias cannot be NULL or empty");
             throw new InvalidValueException(FILTER_PATH_NOT_NULL);
         }
 
         if (argumentCount == ArgumentCount.MANY) {
-            Preconditions.checkState(!params.isEmpty());
+            Preconditions.checkState(!predicate.getParameters().isEmpty());
```

Comment:
Maybe `predicate.getParameters()` can be extracted in a variable and then reused everywhere?